src/compiler/parser.ts

```diff
@@ -2813,7 +2820,7 @@ namespace ts {
             node.parameters.end = parameter.end;
 
             node.equalsGreaterThanToken = parseExpectedToken(SyntaxKind.EqualsGreaterThanToken, /*reportAtCurrentPosition*/ false, Diagnostics._0_expected, "=>");
-            node.body = parseArrowFunctionExpressionBody(/*isAsync*/ false);
+            node.body = parseArrowFunctionExpressionBody(/*isAsync*/ asyncModifier ? true : false);

```

Comment:
do we usually use `!!asyncModifier` instead of `? true : fase`? I don't know which is standard in our code base. I'd prefer !! just because it's shorter.